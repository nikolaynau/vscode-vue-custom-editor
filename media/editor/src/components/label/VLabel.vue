<script setup lang="ts">
import { computed, toRefs } from 'vue';

export interface Props {
  tag?: string;
  truncate?: boolean;
}

const props = withDefaults(defineProps<Props>(), {
  tag: 'label',
  truncate: false
});

const { truncate } = toRefs(props);

const cssClasses = computed(() => ({
  'v-label--truncate': truncate.value
}));
</script>

<template>
  <component class="v-label" :class="cssClasses" :is="tag">
    <slot></slot>
  </component>
</template>

<style>
@import 'label';
</style>
