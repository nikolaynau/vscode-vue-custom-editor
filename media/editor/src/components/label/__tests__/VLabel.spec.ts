import { describe, it, expect } from 'vitest';
import { mount } from '@vue/test-utils';
import VLabel from '../VLabel.vue';

describe('VLabel', () => {
  it('default render props', () => {
    const wrapper = mount(VLabel);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('tag prop', () => {
    const wrapper = mount(VLabel, { props: { tag: 'span' } });
    expect(wrapper.element.tagName).toBe('SPAN');
  });

  it('truncate prop', () => {
    const wrapper = mount(VLabel, { props: { truncate: true } });
    expect(wrapper.classes('v-label--truncate')).toBe(true);
  });

  it('default slot', () => {
    const wrapper = mount(VLabel, { slots: { default: 'Label text' } });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
