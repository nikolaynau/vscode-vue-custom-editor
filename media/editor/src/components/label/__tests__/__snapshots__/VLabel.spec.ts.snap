// Vitest Snapshot v1

exports[`VLabel > default slot 1`] = `"<label class=\\"v-label\\">Label text</label>"`;

exports[`VLabel > render default props 1`] = `"<label class=\\"v-label\\"></label>"`;
